StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/STPApplePayContext+PaymentSheet.swift

```diff
@@ -46,7 +46,33 @@ extension STPApplePayContext {
             paymentInformation: PKPayment,
             completion: @escaping STPIntentClientSecretCompletionBlock
         ) {
-            completion(clientSecret, nil)
+            switch intent {
+            case .paymentIntent(let paymentIntent):
+                completion(paymentIntent.clientSecret, nil)
+            case .setupIntent(let setupIntent):
+                completion(setupIntent.clientSecret, nil)
+            case .deferredIntent(_, let intentConfig):
+                if let confirmHandler = intentConfig.confirmHandler {
+                    confirmHandler(paymentMethod.id, { result in
+                        switch result {
+                        case .success(let clientSecret):
+                            completion(clientSecret, nil)
+                        case .failure(let error):
+                            completion(nil, error)
+                        }
+                    })
+                } else if let serverSideConfirmHandler = intentConfig.confirmHandlerForServerSideConfirmation {
+                    // Don't save Apple Pay PMs
```

Comment:
This comment threw me off, I don't think future readers will understand what this is referring to.

We could do `let shouldSavePaymentMethod = false // The customer isn't requesting to save the payment method`